Use mip start solution

lgottwald opened the pull request:
includes set-get-iterate branch

jajhall reviewed (approved):
I'm happy with this if you are!

lgottwald commented:
I also changes a few other minor things. I.e. for the `HighsBasis` struct the `HighsBasisStatus` was unnecessarily large using the default enum size, usually 4 bytes. I changed it to be 1 byte. Also added some missing `const` to the postsolve stack and fixed one situation that is technically undefined behavior where memcpy with zero size and `NULL` pointer was called.

lgottwald commented:
I think I am happy. Running little more tests, but would probably merge once the CI tests run through
